stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -184,6 +184,10 @@ end
     return x
 end
 
+Base._reverse(A::Bidiagonal, dims) = reverse!(Matrix(A); dims)
+Base._reverse(A::Bidiagonal, dims::Colon) = Bidiagonal(reverse(A.dv), reverse(A.ev), A.uplo)
```

Comment:
You need to reverse the `uplo`, no?
```suggestion
Base._reverse(A::Bidiagonal, dims::Colon) = Bidiagonal(reverse(A.dv), reverse(A.ev), A.uplo == 'U' ? 'L' : 'U')
```